oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -97,9 +107,48 @@ private static ObjectBuilder<TypeMapping> loadMappings(@NotNull TypeMapping.Buil
         if (indexDefinition.inferenceDefinition != null) {
             mapInferenceDefinition(builder, indexDefinition.inferenceDefinition);
         }
+        // We only add mappings if both the inference config (in queryEngineSettings config) and the inference index config are enabled.
+        if (InferenceConfig.getInstance().isInferenceEnabled() && InferenceConfig.getInstance().isEnabled()) {
+            mapInferenceConfig(builder, indexDefinition, InferenceConfig.getInstance());
+        }
         return builder;
     }
 
+    private static void mapInferenceConfig(TypeMapping.Builder builder, @NotNull ElasticIndexDefinition indexDefinition, @NotNull InferenceConfig inferenceConfig) {
+        String indexName = PathUtils.getName(indexDefinition.getIndexName());
+
+        InferenceIndexConfig inferenceIndexConfig = inferenceConfig.getInferenceIndexConfig(indexName);
+        if (InferenceIndexConfig.NOOP.equals(inferenceIndexConfig)) {
+            return;
+        }
+        try {
+            // We are already validating the enricherConfigJson in the InferenceIndexConfig constructor
+            Map<String, Object> enricherConfigJson = mapper.readValue(inferenceConfig.getInferenceIndexConfig(indexName).getEnricherConfig(),
+                    new TypeReference<Map<String, Object>>() {
+                    });
+            // Store the enricher configuration in the index metadata so that it can be used by the enricher service
+            enricherConfigJson.forEach((k, v) -> {
+                builder.meta(k, JsonData.of(v));
+            });
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException("enricherConfig parsing should never fail as it is validated in InferenceIndexConfig" + e.getMessage());
+        }
+
+        builder.properties(InferenceConstants.VECTOR_SPACES, b -> b.object(spaces -> {
+            for (var inferenceModelConfig : inferenceIndexConfig.getInferenceModelConfigs().entrySet()) {
+                if (inferenceModelConfig.getValue().isEnabled()) {
+                    spaces.properties(inferenceModelConfig.getKey(), v -> v.nested(vb -> {
+                        vb.properties("id", p -> p.keyword(k -> k));
+                        vb.properties("vector", p -> p.denseVector(dv -> dv));
+                        vb.properties("metadata", p -> p.object(o -> o.enabled(false)));
+                        return vb;
+                    }));
+                }
+            }
+            return spaces;
+        }));
+    }
```

Comment:
I don't think we should do this. We need a way to abstract the mapping and make it configurable.
One possibility is to have an elastic-specific index definition property that contains the actual mapping that will be sent to ES at index creation. The same approach can be used for the :enrich field. In that case, we can should also specify the default value (eg: PENDING) that will be sent out every time a doc gets updated. These changes are actually not related to the inference support but are generic. Can we create a separate PR for that?